shared/params/loader.go

```diff
@@ -30,7 +30,7 @@ func LoadChainConfigFile(chainConfigFileName string) {
 	}
 	yamlFile = []byte(strings.Join(lines, "\n"))
 	conf := MainnetConfig()
-	if err := yaml.Unmarshal(yamlFile, conf); err != nil {
+	if err := yaml.UnmarshalStrict(yamlFile, conf); err != nil {
 		log.WithError(err).Fatal("Failed to parse chain config yaml file.")
```

Comment:
Consider checking this error. If it's a type error, it shouldn't be fatal.

```suggestion
	if err := yaml.UnmarshalStrict(yamlFile, conf); err != nil {
		if _, ok := err.(yaml.TypeError); !ok {
			log.WithError(err).Fatal("Failed to parse chain config yaml file.")
		} else {
			log.WithError(err).Error("There were some issues parsing the config from a yaml file")
		}
```